doc/content/getting-started/installation/configuration/_index.md

```diff
@@ -141,3 +143,56 @@ Below is an example `ttn-lw-stack-docker.yml` file for the Enterprise stack:
 {{< /highlight >}}
 
 > NOTE: Make note of the client secret, as it will be needed again when initializing {{% tts %}}.
+
+## Localhost
+
+If you are configuring and running {{% tts %}} on a local machine with no public IP or DNS address, some special considerations must be made. `localhost` has a different meaning on your machine than inside the Docker container where {{% tts %}} runs, which can cause problems if you use `localhost` in your configuration.
+
+`localhost` addresses on your machine will resolve to your machine, while `localhost` inside the Docker container will resolve inside the Docker container. In `docker-compose.yml`, we forward requests on ports 80 and 443 to ports 1885 and 8885 inside the container, so visiting `localhost` (which is really `localhost:80`) on your machine actually takes you to `localhost:1885` within the container.
+
+If you configure your `is.oauth.ui.canonical-url` as `localhost`, this causes `Token Exchange Refused` errors when you try to log in to the console, because an authorization request generated from within the container will not be redirected to port 1885 or 8885, where {{% tts %}} is listening.
+
+### Solution 1: Use the IP address of your computer on your local network
+
+The best solution is to use the IP address of your machine on your local network so that redirects from your machine, from the docker container, or from anywhere inside your local network all resolve at the same place, on your machine. Follow instructions [here](https://www.avast.com/c-how-to-find-ip-address#topic-4) to find the local IP address of your computer. Use that IP address as your server address, i.e replace `thethings.example.com` with that IP address. You may also generate a self-signed certificate for that IP address by following instructions in [certificates]({{< relref "certificates" >}}).
+
+This will still allow you to see the console by entering `localhost` or your local IP in your browser. It will also allow you to connect to {{% tts %}} from any machine inside your local network.
```

Comment:
```suggestion
This will still allow you to see {{% tts %}} Console by entering `localhost` or your local IP in your browser. It will also allow you to connect to {{% tts %}} from any machine inside your local network.
```

Reply:
I don't think the last sentence is true, as public addresses are still `localhost`

Reply:
> I don't think the last sentence is true, as public addresses are still `localhost`

What do you mean by public addresses?

Reply:
https://thethingsstack.io/getting-started/installation/configuration/#component-urls

Reply:
But the point of this solution is to use your local IP and not use localhost

Reply:
OK I get it